salsah/src/public/js/jquery.searchbox.js

```diff
@@ -206,7 +208,12 @@
 							left: pos.left,
 							top: pos.top
 						}).show();
-						$this.val('');
+						// if the input already has a value and it is valid, back it up
```

Comment:
exactly.

Do you mean that instead of using cached values, I should go and fetch the source of this value? Yes, that would be smarter.

But when leaving the scope on `jquery.searchbox.js:212` finding that value was not obvious to me.